components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIKeyValidator.java

```diff
@@ -468,53 +480,67 @@ public String getResourceAuthenticationScheme(MessageContext synCtx) throws APIS
         return APIConstants.NO_MATCHING_AUTH_SCHEME;*/
     }
 
-    public VerbInfoDTO findMatchingVerb(MessageContext synCtx) throws ResourceNotFoundException, APISecurityException {
-
-        VerbInfoDTO verb = null;
-        String resourceCacheKey = null;
+    public List<VerbInfoDTO> findMatchingVerb(MessageContext synCtx) throws ResourceNotFoundException, APISecurityException {
 
-        //This function is used by more than one handler. If on one execution of this function, it has found and placed
-        //the matching verb in the cache, the same can be re-used from all handlers since all handlers share the same
-        //MessageContext. The API_RESOURCE_CACHE_KEY property will be set in the MessageContext to indicate that the
-        //verb has been put into the cache.
-        if (isGatewayAPIResourceValidationEnabled) {
-            resourceCacheKey = (String) synCtx.getProperty(APIConstants.API_RESOURCE_CACHE_KEY);
-            if (resourceCacheKey != null) {
-                verb = (VerbInfoDTO) getResourceCache().get(resourceCacheKey);
-                //Cache hit
-                if (verb != null) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Found resource in Cache for key: " + resourceCacheKey);
-                    }
-                    return verb;
-                }
-                if (log.isDebugEnabled()) {
-                    log.debug("Resource not found in cache for key: " + resourceCacheKey);
-                }
-            }
-        }
-        String resourceString = (String) synCtx.getProperty(APIConstants.API_ELECTED_RESOURCE);
+        List<VerbInfoDTO>  verbInfoList =  new ArrayList<>();
+        String resourceCacheKey;
+        String httpMethod = (String) ((Axis2MessageContext) synCtx).getAxis2MessageContext().
+                getProperty(Constants.Configuration.HTTP_METHOD);
         String apiContext = (String) synCtx.getProperty(RESTConstants.REST_API_CONTEXT);
         String apiVersion = (String) synCtx.getProperty(RESTConstants.SYNAPSE_REST_API_VERSION);
         String fullRequestPath = (String) synCtx.getProperty(RESTConstants.REST_FULL_REQUEST_PATH);
         String apiName = (String) synCtx.getProperty(RESTConstants.SYNAPSE_REST_API);
 
+        String electedResource = (String) synCtx.getProperty(APIConstants.API_ELECTED_RESOURCE);
+        ArrayList<String> resourceArray = null;
+
+        if (electedResource != null) {
+            resourceArray = new ArrayList<>
+                    (Arrays.asList(electedResource.split(",")));
```

Comment:
Doesn't seem exceed 120 line break. Can go in one liner?

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80